tests/virt_autotest/validate_host_and_guests.pm

```diff
@@ -0,0 +1,48 @@
+# SUSE's openQA tests
+#
+# Copyright 2012-2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Do a basic examination to the host and guests after tests run
+# Maintainer: qe-virt@suse.de
+
+package validate_host_and_guests;
+
+use strict;
+use warnings;
+use base "virt_feature_test_base";
+use testapi;
+use virt_autotest::common;
+use virt_autotest::utils qw(check_host_health check_guest_health);
+
+sub run_test {
+
+    my $failures = check_host_health;
+    die "Host is not healthy: $failures" unless $failures eq "";
+    foreach my $guest (keys %virt_autotest::common::guests) {
```

Comment:
Adopt ```check_port_state``` for guest as well before trying to connect to it. This will save time and decrease uncertainty.

Reply:
If guest is not reachable, exclude it from logs collecting. So only  passing the others to ```virt_utils::collect_host_and_guest_logs;```